### PR TITLE
DTS-29397 hide tools and connections based on flag

### PR DESCRIPTION
**Pull request by brahmanand1**

### CODE REVIEW

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/appsetting/rest/GlobalFieldsController.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Controller for global fields
+ *
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/appsetting/rest/GlobalFieldsController.java:33]: add author name

**Comment by Chittauri** [comment]
@kunkambl  use "/api/configDetails" instead of creating new data class.

**Review by Chittauri — APPROVED** [review]

**Review by shunaray — CHANGES_REQUESTED** [review]
Also make db changes for installation scripts both mongock & database_scripts

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/mongock/rollback/release_810/RepoToolProviderUrlChange.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
+package com.publicissapient.kpidashboard.apis.mongock.rollback.release_810;
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/mongock/rollback/release_810/RepoToolProviderUrlChange.java:1]: add copyright

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/mongock/rollback/release_810/RepoToolProviderUrlChange.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+import io.mongock.api.annotations.Execution;
+import io.mongock.api.annotations.RollbackExecution;
+import org.bson.Document;
+import org.springframework.data.mongodb.core.MongoTemplate;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@ChangeUnit(id = "r_repo_tool_provider_url_change", order = "08110", author = "kunkambl", systemVersion = "8.1.0")
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/mongock/rollback/release_810/RepoToolProviderUrlChange.java:16]: use suppress warning, add author comment

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/mongock/upgrade/release_810/RepoToolProviderUrlChange.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
+package com.publicissapient.kpidashboard.apis.mongock.upgrade.release_810;
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/mongock/upgrade/release_810/RepoToolProviderUrlChange.java:1]: add copyright, author and suppress warning

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/appsetting/service/ProcessorServiceImpl.java [hunk 5]
```diff
@@ -73,13 +74,28 @@ public class ProcessorServiceImpl implements ProcessorService {
...
+	private CustomApiConfig customApiConfig;
 
 	@Override
 	public ServiceResponse getAllProcessorDetails() {
 		List<Processor> listProcessor = new ArrayList<>();
+		Boolean repoToolFlag = customApiConfig.getIsRepoToolEnable();
 		processorRepository.findAll().iterator().forEachRemaining(p -> {
 			if (null != p) {
-				listProcessor.add(p);
+				String processorName = p.getProcessorName();
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/appsetting/service/ProcessorServiceImpl.java:87]: optimize the check

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/connection/service/ConnectionServiceImpl.java [hunk 6]
```diff
@@ -169,6 +177,15 @@ public ServiceResponse getConnectionByType(String type) {
 		return new ServiceResponse(true, "Found type@" + type, typeList);
 	}
 
+	private List<Connection> getConnectionList(String type) {
+		if (Boolean.TRUE.equals(customApiConfig.getIsRepoToolEnable()) && type.equalsIgnoreCase(TOOL_GITHUB)) {
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/connection/service/ConnectionServiceImpl.java:181]: add todo--debbie

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/appsetting/service/ProcessorServiceImpl.java [hunk 7]
```diff
@@ -73,13 +74,27 @@ public class ProcessorServiceImpl implements ProcessorService {
...
+				boolean isRepoTool = processorName.equalsIgnoreCase(ProcessorConstants.REPO_TOOLS);
+				boolean shouldAddToList = (repoToolFlag.equals(Boolean.FALSE) && !isRepoTool)
+						|| (repoToolFlag.equals(Boolean.TRUE)
+								&& !processorName.equalsIgnoreCase(ProcessorConstants.GITLAB)
+								&& !processorName.equalsIgnoreCase(ProcessorConstants.GITHUB)
+								&& !processorName.equalsIgnoreCase(ProcessorConstants.BITBUCKET)
+								&& !processorName.equalsIgnoreCase(ProcessorConstants.AZUREREPO));
+				if (shouldAddToList) {
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/appsetting/service/ProcessorServiceImpl.java:95]: boolean toggle= true;
toggle= toggle && !processorName.equalsIgnoreCase(ProcessorConstants.GITLAB);

toggle= toggle && !processorName.equalsIgnoreCase(ProcessorConstants.GITHUB);....... use this

**Review by Chittauri — APPROVED** [review]

**Review by shunaray — APPROVED** [review]

**Pull request merged**